### PR TITLE
Add release for 10.1.0

### PR DESCRIPTION
**Pull request by tuommaki**
10.1.0 brings the Node Pools release.

### CODE REVIEW

**Comment by tuommaki** [comment]
@J-K-C Do you write release notes to `release-notes` folder? See https://github.com/giantswarm/releases/blob/master/release-notes/aws/v9.0.0.md

File: aws.yaml [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
       version: 0.7.0
     - name: cluster-operator
       provider: aws
-      version: 2.0.0-dev
-  date: 2019-12-05T12:00:00Z
-  version: 10.0.0
+      version: 2.0.0
+  date: 2019-12-18T14:00:00Z
+  version: 10.1.0
 - active: false
```
Comment (kopiczko) [inline aws.yaml]: You'll also need to add a new release with aws-operator@8.0.1-dev and cluster-operator@2.0.1-dev for testing. Do you want to provide that in a separate PR?

Comment (xh3b4sd) [reply aws.yaml]: We can add this here before merging.

**Comment by marians** [comment]
FYI: more changes by @Oshratn and myself incoming for the release notes.

File: release-notes/aws/v10.1.0.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
+# :zap: Giant Swarm Release 10.1.0 for AWS is now active for you! :zap:
+
+We are pleased to announce that the new Cluster API enabled release v10.1.0 contains the much anticipated Node Pools feature.
```
Comment (teemow) [inline release-notes/aws/v10.1.0.md]:
```suggestion
We are pleased to announce that the new release v10.1.0 contains the much anticipated Node Pools feature.
```

Comment (teemow) [reply release-notes/aws/v10.1.0.md]: Cluster API is so far irrelevant for the users. Happy to have this in the changelog. But only confuses here imo.

**Comment by teemow** [comment]
LGTM besides the small nitpick with the cluster api.

File: release-notes/aws/v10.1.0.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+We want to remind you that the Node Pools enabled releases imply a breaking change. This means **existing clusters** below the 10.x.x major releases **are not upgradable**. 
+We will continue to support the 9.x.x releases with bug fixes and security patches for the next 6 months. During that timeframe please **ensure you  move all your workloads to new clusters**. We are at your service to assist with this.
```
Comment (puja108) [inline release-notes/aws/v10.1.0.md]:
```suggestion
We will continue to support the 9.x.x releases with bug fixes and security patches for the next 6 months. During that timeframe please **ensure that you move all your workloads to new clusters**. We are at your service to assist with this.
```

**Comment by puja108** [comment]
LGTM from me too, would have been good to be pinged on this, as team batman was significantly involved in one of the bigger changes here, I only by accident found this as it was mentioned in a closed PR and I had that still open.

File: release-notes/aws/v10.1.0.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+*Please note*: With this release, we are NO LONGER rolling out NGINX Ingress Controller by default. It is now an optional App in the App Catalog and can be installed and configured on-demand. Or you can choose your own.
```
Comment (puja108) [inline release-notes/aws/v10.1.0.md]: I am sorry for this last minute notice, but we just realized that optional IC is blocked in China for the lack of route53. We have a plan to fix this, but for now we decided to go with a notice that with 10.1.0 the optional IC will not be available in China and we would recommend that customers in China wait for our fixes for China to land in a soon to be released patch version before they start clusters with 10.x. cc @teemow you fine with that?

**Comment by marians** [comment]
Added @puja108's paragraph and modified the formatting to make the two caveats stand out as bullet items. Also replaced CAPITALIZATION with _italic_ emphasis.

File: release-notes/aws/v10.1.0.md (hunk 4 above)
Comment (marians) [reply release-notes/aws/v10.1.0.md]: Is this resolved with the latest addition https://github.com/giantswarm/releases/pull/45/commits/72f7e50f4225f0c9e58c2fac5270794ccbe69beb ?

**Pull request merged**